homework1/main.c

```diff
@@ -0,0 +1,311 @@
+            int divisible = 0;
+            int divider = 0;
+
+            printf("Enter divisible and divider: ");
+            scanf("%d%d", &divisible, &divider);
+
+            printf("Incomplete quotent in division: %d/%d = %d\n\n", divisible, divider, task2(divisible, divider));
+
+
+            break;
+        }
+        case 3: {
+            int array[arraySize] = { 0 };
+            int firstIndex = 0;
+            int secondIndex = 0;
+            int arrayLength = 0;
+
+
+            bool isIncorrectInput = true;
+            while (isIncorrectInput) {
+                printf("Enter lengths of two parts of the array: ");
+                scanf("%d%d", &firstIndex, &secondIndex);
+                if (firstIndex < 0 || secondIndex < 0) isIncorrectInput = true;
+                else isIncorrectInput = false;
+                if (isIncorrectInput) printf("Incorrect input. Both numbers aren't negative. Try again!\n");
+            }
+
+
+            arrayLength = firstIndex + secondIndex;
+
+
+            printf("Enter the array: ");
+
+            for (int i = 0; i < arrayLength; ++i) {
+                scanf("%d", &array[i]);
+            }
+
+            reverseArray(array, firstIndex); //переворот первой части
+            reverseArray(array, arrayLength); //переворот всего массива
+            reverseArray(array, secondIndex); //переворот второй части
+
+
+
+            for (int i = 0; i < arrayLength; ++i) {
+                printf("%d ", array[i]);
+            }
+            printf("\n\n");
+
+
+            break;
+        }
+        case 4: {
+            int luckyTicketsAmount = 0;
+
+            luckyTicketsAmount = task4();
+            printf("Amount of lucky tickets with length 6 is %d\n\n", luckyTicketsAmount);
+            break;
+        }
+        case 5: {
+            char string[100] = "";
+
+            printf("Enter the sequence: ");
+            scanf("%s", string, 100);
+
+            if (task5(string)) {
+                printf("Bracket sequence is correct\n\n");
+            }
+            else {
+                printf("Bracket sequence is incorrect\n\n");
+            }
+
+
+
+            break;
+        }
+        case 6: {
+            char string[100] = "";
+            char substring[100] = "";
+
+
+            printf("Enter the string (<100 symbols): ");
+            scanf("%s", string, 100);
+            printf("Enter the substring (<100 symbols): ");
+            scanf("%s", substring, 100);
+
+            printf("Amount of entries of substring in string is %d\n\n", task6(string, substring));
+
+
+            break;
+        }
+        case 7: {
+            int number = 0;
+            bool arrayPrimeNumbers[arraySize] = { false };
+
+
+
+            printf("Enter your number (0 < x < 1000): ");
+            scanf("%d", &number);
+
+            if (number < 0) {
+                printf("Your number is negative, try again\n\n");
+                break;
+            }
+
+            if (number > 1000) {
```

Comment:
```suggestion
            if (number >= 1000) {
```
Иначе выход за пределы массива